externalpolicy/text_language/text_language.go

```diff
@@ -85,287 +82,189 @@ func (p *TextConstraintLanguagePlugin) Validate(dconstraints interface{}) (bool,
 // This function parses out the next property expression and returns it along with the remainder of the expression.
 // It returns, the parsed out expression, and the remainder of the full expression, or an error.
 func (p *TextConstraintLanguagePlugin) GetNextExpression(expression string) (string, string, error) {
-
 	// The input expression string should begin with an expression that can be captured and returned, or it is empty.
 	// This should be true because the full expression should have been validated before calling this function.
-
-	if len(expression) == 0 {
+	if strings.TrimSpace(expression) == "" {
 		return "", "", nil
 	}
 
-	// Split the expression based on whitespace in the string.
-	pieces := strings.Split(expression, " ")
-	if len(pieces) < 3 {
-		return "", "", errors.New(i18n.GetMessagePrinter().Sprintf("found %v token(s), expecting 3 in an expression %v, expected form is <property> == <value>", len(pieces), expression))
-	}
-	quote := "\""
-	value := pieces[2]
-	remainder := strings.Join(pieces[3:], " ")
-	if strings.Count(pieces[2], quote)%2 == 1 {
-		for i, piece := range pieces[3:] {
-			value = fmt.Sprintf("%s %s", value, piece)
-			//fmt.Println(value)
-			if strings.Contains(piece, quote) {
-				remainder = strings.Join(pieces[i+4:], " ")
-				break
+	lexDef := getLexer()
+	def := getLexer().Symbols()
+	lex, err := lexDef.Lex(strings.NewReader(expression))
+	if err != nil {
+		return "", expression, err
+	}
+	nextToken, err := lex.Next()
+	if err != nil {
+		return "", expression, err
+	}
+	nextRune := nextToken.Type
+	// Start of property expression. This case will consume the entire expression.
+	if nextRune == def["Str"] {
+		name := nextToken.Value
+		var opType rune
+		var valType rune
+		op := ""
+		val := ""
+		nextToken, err = lex.Next()
+		if err != nil {
+			return "", expression, fmt.Errorf("Unrecognized token found: %v", err)
```

Comment:
What causes this error to occur?

Reply:
When the lexer encounters something in the constraint that does not match any of  the specified tokens.